external/defects4c/evaluation/run_tests.py

```
@@ -0,0 +1,105 @@
+
+DEFAULT_TIMEOUT = 60
+
+
+def _run_cmd(cmd: str, cwd: str, trusted: bool, label: str) -> int:
+    """Run a build or test command with optional shell=True and a timeout."""
+    if not cmd:
+        print(f"{label} command is empty", file=sys.stderr)
+        return 1
+    if trusted:
+        # Trusted mode: allow shell=True but still enforce a timeout.
+        result = subprocess.run(cmd, shell=True, cwd=cwd, timeout=DEFAULT_TIMEOUT)
```

Ruff (0.15.2)

[error] `subprocess` call with `shell=True` identified, security issue

(S602)